db/migrate/20230724094715_add_user_to_books.rb

```diff
@@ -0,0 +1,5 @@
+class AddUserToBooks < ActiveRecord::Migration[7.0]
+  def change
+    add_reference :books, :user, null: false, foreign_key: true,default: 1
```

Comment:
Normally we will not specify the default value to foreign key, in this case, if the old records break nullable constraint, we can define it as `null: true` first and then migrate the old data, and eventually set `null: false`

```ruby
add_reference :books, :user, null: true, foreign_key: true

user = User.find_or_create_by!(email: 'kasawat.c@datawow.io')

Book.find_each do |book|
  book.user = user
  book.save!
end

change_column_null :books, user_id, false # please check syntax again not sure if it is correct
```

In this case, we just dump all the books to one user, but actually we need to find some logics to define the owner of each books.